src/index.ts

```diff
@@ -189,10 +217,15 @@ function useDetectScroll(props: ScrollProps = {}): ScrollInfo {
       }
     };
 
-    window.addEventListener('scroll', onScroll);
+    target instanceof Window
```

Comment:
You can do the same as the previous `useEffect` to prevent ternary operation on event listeners.